extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -424,10 +424,18 @@ class QueueProcessor {
     // Object metadata may contain multiple elements for a single part if
     // the part was originally copied from another MPU. Here we reduce the
     // locations array to a single element for each part.
-    _getLocations(sourceEntry) {
+    _getLocations(sourceEntry, log) {
         const locations = sourceEntry.getLocation();
         const reducedLocations = [];
         let partTotal = 0;
+        if (locations.some(
```

Comment:
Could we add this check to the for loop below? That way we avoid looping through a second time.